packages/turbo-codemod/src/transforms/create-turbo-config.ts

```diff
@@ -1,64 +1,70 @@
-        }
-        skip("turbo.json", chalk.dim("(dry run)"));
-        if (flags.print) {
-          console.log(JSON.stringify(remainingPkgJson, null, 2));
-        }
-        skip("package.json", chalk.dim("(dry run)"));
-        skippedCount += 2;
-      } else {
-        if (flags.print) {
-          console.log(JSON.stringify(turboConfig, null, 2));
-        }
-        ok("turbo.json", chalk.dim("(created)"));
-        fs.writeJsonSync(turboConfigPath, turboConfig, { spaces: 2 });
-        if (flags.print) {
-          console.log(JSON.stringify(remainingPkgJson, null, 2));
-        }
-        ok("package.json", chalk.dim("(remove turbo key)"));
-        fs.writeJsonSync(rootPackageJsonPath, remainingPkgJson, { spaces: 2 });
-        modifiedCount += 2;
-        unmodifiedCount -= 2;
-      }
-    } else {
-      error('"turbo" key does not exist in "package.json"');
-      process.exit(1);
-    }
-    console.log("All done.");
-    console.log("Results:");
-    console.log(chalk.red(`0 errors`));
-    console.log(chalk.yellow(`${skippedCount} skipped`));
-    console.log(chalk.yellow(`${unmodifiedCount} unmodified`));
-    console.log(chalk.green(`${modifiedCount} modified`));
+// transformer details
+const TRANSFORMER = "create-turbo-config";
+const DESCRIPTION =
+  'Create the `turbo.json` file from an existing "turbo" key in `package.json`';
+const INTRODUCED_IN = "1.1.0";
+
+export function transformer({
+  root,
+  options,
+}: TransformerArgs): TransformerResults {
+  const { log, runner } = getTransformerHelpers({
+    transformer: TRANSFORMER,
+    rootPath: root,
+    options,
+  });
+
+  log.info(`Migrating "package.json" "turbo" key to "turbo.json" file...`);
+  const turboConfigPath = path.join(root, "turbo.json");
+  const rootPackageJsonPath = path.join(root, "package.json");
+  if (!fs.existsSync(rootPackageJsonPath)) {
+    return runner.abortTransform({
+      reason: `No package.json found at ${root}. Is the path correct?`,
+    });
+  }
+
+  // read files
+  const rootPackageJson = fs.readJsonSync(rootPackageJsonPath);
+  let rootTurboJson = null;
+  try {
+    rootTurboJson = fs.readJSONSync(turboConfigPath);
+  } catch (err) {
+    rootTurboJson = null;
+  }
+
+  // modify files
+  let transformedPackageJson = rootPackageJson;
+  let transformedTurboConfig = rootTurboJson;
+  if (!rootTurboJson && rootPackageJson["turbo"]) {
+    const { turbo: turboConfig, ...remainingPkgJson } = rootPackageJson;
+    transformedTurboConfig = turboConfig;
+    transformedPackageJson = remainingPkgJson;
   }
+
+  runner.modifyFile({
+    filePath: turboConfigPath,
+    after: transformedTurboConfig,
+  });
+  runner.modifyFile({
+    filePath: rootPackageJsonPath,
+    after: transformedPackageJson,
+  });
+
+  return runner.finish();
 }
+
+const transformerMeta = {
+  name: `${TRANSFORMER}: ${DESCRIPTION}`,
+  value: TRANSFORMER,
+  introducedIn: INTRODUCED_IN,
+  transformer,
+};
+
+export default transformerMeta;
```

Comment:
bit weird to export an object as the default export (reminiscent of CJS). Could also do named exports for each of the keys:

```
export const name = `TRANSFORMER...`
export const value = TRANSFORMER
// etc
```

Reply:
This is done for simpler type safety. Because we dynamically require the transform files, we establish an API that a transformer _must_ satisfy. A single export of the `Transform` type:

```
export type Transformer = {
  name: string;
  value: string;
  introducedIn: string;
  transformer: (args: TransformerArgs) => TransformerResults;
};
```